perf/src/main/java/tachyon/perf/benchmark/massive/MassiveThread.java

```diff
@@ -0,0 +1,185 @@
+    String syncFileName = mTaskId + "-" + mId;
+    mFileSystem.delete(syncDirPath + "/" + syncFileName, false);
+    while (!mFileSystem.listFullPath(syncDirPath).isEmpty()) {
+      try {
+        Thread.sleep(300);
+      } catch (InterruptedException e) {
+        LOG.error("Error in Sync Barrier", e);
+      }
+    }
+  }
+
+  @Override
+  public void run() {
+    long basicBytes = 0;
+    long basicTimeMs = 0;
+    long readBytes = 0;
+    long readTimeMs = 0;
+    long writeBytes = 0;
+    long writeTimeMs = 0;
+    mSuccess = true;
+    String tmpDir = mWorkDir + "/tmp";
+    String dataDir;
+    if (mShuffle) {
+      dataDir = mWorkDir + "/data";
+    } else {
+      dataDir = mWorkDir + "/data/" + mTaskId;
+    }
+
+    long tTimeMs = System.currentTimeMillis();
+    for (int b = 0; b < mBasicFilesNum; b ++) {
+      try {
+        String fileName = mTaskId + "-" + mId + "-" + b;
+        Operators.writeSingleFile(mFileSystem, dataDir + "/" + fileName, mFileLength, mBufferSize);
+        basicBytes += mFileLength;
+      } catch (IOException e) {
+        LOG.error("Failed to write basic file", e);
+        mSuccess = false;
+        break;
+      }
+    }
+    tTimeMs = System.currentTimeMillis() - tTimeMs;
+    basicTimeMs += tTimeMs;
+
+    try {
+      syncBarrier();
+    } catch (IOException e) {
+      LOG.error("Error in Sync Barrier", e);
+      mSuccess = false;
+    }
+    int index = 0;
+    long limitTimeMs = System.currentTimeMillis() + mLimitTimeMs;
+    while ((tTimeMs = System.currentTimeMillis()) < limitTimeMs) {
+      if (mRand.nextBoolean()) { // read
+        try {
+          List<String> candidates = mFileSystem.listFullPath(dataDir);
+          String readFilePath = ListGenerator.generateRandomReadFiles(1, candidates).get(0);
+          readBytes += Operators.readSingleFile(mFileSystem, readFilePath, mBufferSize);
+        } catch (IOException e) {
+          LOG.error("Failed to read file", e);
+          mSuccess = false;
+        }
+        readTimeMs += System.currentTimeMillis() - tTimeMs;
+      } else { // write
+        try {
+          String writeFileName = mTaskId + "-" + mId + "--" + index;
+          Operators.writeSingleFile(mFileSystem, tmpDir + "/" + writeFileName, mFileLength,
+              mBufferSize);
+          writeBytes += mFileLength;
+          mFileSystem.rename(tmpDir + "/" + writeFileName, dataDir + "/" + writeFileName);
+        } catch (IOException e) {
+          LOG.error("Failed to write file", e);
+          mSuccess = false;
+        }
+        writeTimeMs += System.currentTimeMillis() - tTimeMs;
+      }
+      index ++;
+    }
+
+    mBasicWriteThroughput =
+        (basicBytes == 0) ? 0 : (basicBytes / 1024.0 / 1024.0) / (basicTimeMs / 1000.0);
+    mReadThroughput = (readBytes == 0) ? 0 : (readBytes / 1024.0 / 1024.0) / (readTimeMs / 1000.0);
+    mWriteThroughput =
+        (writeBytes == 0) ? 0 : (writeBytes / 1024.0 / 1024.0) / (writeTimeMs / 1000.0);
```

Comment:
Should the check here be for `TimeMs == 0`? Since the result will be 0 if `Bytes == 0` anyway?